lamindb/_artifact.py

```diff
@@ -461,23 +473,28 @@ def data_is_mudata(data: MuData | UPathStr):
 
         if isinstance(data, MuData):
             return True
-    if isinstance(data, (str, Path, UPath)):
-        return Path(data).suffix in {".h5mu"}
+    if isinstance(data, (str, Path)):
+        return UPath(data).suffix in {".h5mu"}
     return False
 
 
 def _check_accessor_artifact(data: Any, accessor: str | None = None):
-    if accessor is None and not isinstance(data, (str, Path, UPath)):
+    if accessor is None:
         if isinstance(data, pd.DataFrame):
             logger.warning("data is a DataFrame, please use .from_df()")
             accessor = "DataFrame"
-        elif data_is_anndata(data):
-            logger.warning("data is an AnnData, please use .from_anndata()")
+            return accessor
+
+        data_is_path = isinstance(data, (str, Path))
+        if data_is_anndata(data):
+            if not data_is_path:
+                logger.warning("data is an AnnData, please use .from_anndata()")
```

Comment:
@falexwolf `.from_anndata` implies only in-memory object or not? I can see from typing and examples, that only in-memory object is allowed, so i add this suppression of the warning for adatas on disk. Is this correct?

Reply:
Or we also should allow now `Artifact.from_anndata("something.h5ad")`?

Reply:
Only in memory is allowed here!

Reply:
Ok, got it.

Reply:
I think it'd be nice for a few applications if we allowed backed objects, but then typing is going to be a mess again. So, just in memory is easier for now.